### PR TITLE
Bugfix networking code

### PR DESCRIPTION
**Pull request by zpqrtbnk**
Depends on #512

Kubernetes tests reveal that in some occasion our networking code does not like connections to member going up and down in unpredictable ways - this PR fixes the problem.

### CODE REVIEW

File: src/Hazelcast.Net/Clustering/ClusterConnections.cs [hunk 1]
```diff
@@ -120,10 +127,22 @@ await foreach(var (member, token) in _clusterMembers.MembersToConnect.WithCancel
                 else
                 {
                     var details = wasCanceled ? "canceled" : "failed";
```
Comment (alisengul53) [inline src/Hazelcast.Net/Clustering/ClusterConnections.cs:129]: Is it still needed to set `details` as canceled or failed here?

File: src/Hazelcast.Net.Tests/Clustering/ConnectMembersTests.cs [hunk 2]
```diff
@@ -68,53 +70,66 @@ await foreach (var (member, token) in memberConnectionQueue.WithCancellation(can
...
+
+            await mutex.WaitAsync(); // blocks the connections
 
             queue.Add(MemberInfo(NetworkAddress.Parse("127.0.0.1:4")));
             await Task.Delay(500);
 
-            queue.Suspend();
-            mutex.Release(); // resume => should cancel current connect
+            var task = queue.SuspendAsync();
+            Assert.That(task.IsCompleted, Is.False);
```
Comment (alisengul53) [inline src/Hazelcast.Net.Tests/Clustering/ConnectMembersTests.cs:100]: Is there any specific reason to use always `Assert.That`
What about `Assert.IsTrue(task.IsCompleated)` for true false checks

File: src/Hazelcast.Net/Clustering/ClusterMembers.cs [hunk 3]
```diff
@@ -167,18 +176,25 @@ public void AddConnection(MemberConnection connection, bool isNewCluster)
...
                     {
-                        _logger.LogDebug($"Added connection {connection.Id.ToShortString()} to member {connection.MemberId.ToShortString()}.");
+                        if (_logger.IsEnabled(LogLevel.Debug))
+                        {
+                            var msg = $"Added connection {connection.Id.ToShortString()} to member {connection.MemberId.ToShortString()} at {connection.Address}";
+                            msg += existingConnection == null
+                                ? "."
+                                : $", replacing connection {existingConnection.Id.ToShortString()} at {existingConnection.Address}.";
+                            _logger.LogDebug(msg);
+                        }
```
Comment (alisengul53) [inline src/Hazelcast.Net/Clustering/ClusterMembers.cs:192]: Should we log something here when the LogLevel is not debug?
If it is unneccessary, what about the using `else if` directly

File: src/Hazelcast.Net/Clustering/ClusterConnections.cs (hunk 1 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net/Clustering/ClusterConnections.cs:129]: well, yes, I think so? `exception` could be `null`?

Comment (alisengul53) [reply src/Hazelcast.Net/Clustering/ClusterConnections.cs:129]: I mean in every block `details` var is set to something else, are we using the values `canceled` or `failed` after setting it?

**Review by alisengul53 — APPROVED** [review]
Looks good to me generally!

File: src/Hazelcast.Net/Clustering/ClusterConnections.cs (hunk 1 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net/Clustering/ClusterConnections.cs:129]: Yes but if `exception` is null? None of the blocks run, or am I missing something?

Comment (alisengul53) [reply src/Hazelcast.Net/Clustering/ClusterConnections.cs:129]: Ah sorry my bad, i missed the `details` at the end of the line

**Pull request merged**